asserts/ifacedecls.go

```diff
@@ -98,42 +111,94 @@ func compileMapAttrMatcher(cc compileContext, m map[string]interface{}) (attrMat
 	return matcher, nil
 }
 
-func matchEntry(apath, k string, matcher1 attrMatcher, v interface{}) error {
+func matchEntry(apath, k string, matcher1 attrMatcher, v interface{}, ctx AttrMatchContext) error {
 	apath = chain(apath, k)
-	if v == nil {
+	if _, ok := matcher1.(missingAttrMatcher); !ok && v == nil {
```

Comment:
we could, but given that we decided that only $MISSING should be an exception to the rule that if attribute has a constraint, attribute must be present, I thought it's better to express the exception this way (I could put a clear comment about that though), we can also revisit this once $MISSING is not the only exception anymore for that